src/main/java/duke/command/DeadlineCommand.java

```diff
@@ -0,0 +1,88 @@
+package duke.command;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.Objects;
+
+import duke.DukeException;
+import duke.Storage;
+import duke.task.Deadline;
+import duke.task.TaskList;
+import duke.ui.UI;
+
+/**
+ * Command to create a Deadline Task.
+ */
+public class DeadlineCommand extends NonemptyArgumentCommand implements Command {
+
+    private static final String commandString = "deadline";
+    private final String arguments;
+
+    /**
+     * Constructor for a DeadlineCommand
+     *
+     * @param arguments arguments for the DeadlineCommand
+     */
+    public DeadlineCommand(String arguments) {
+        this.arguments = arguments;
+    }
+
+    /**
+     * Validate arguments to this command.
+     * They must,
+     * 1. Be in the format [description] /by YYYY-MM-DD
+     *
+     * @param arguments argument to validate
+     * @throws DukeException is argument is not valid
+     */
+    @Override
+    protected void validate(String arguments) throws DukeException {
```

Comment:
The logic here is very clean and easy to read especially because each subsequent check is logically following the previous!